src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -161,6 +161,14 @@ extern "C" uint32_t NetSecurityNative_InitSecContext(uint32_t* minorStatus,
     gss_OID desiredMech = isNtlm ? GSS_NTLM_MECHANISM : GSS_SPNEGO_MECHANISM;
 #else
     assert(!isNtlm && "NTLM is not supported by MIT libgssapi_krb5");
+
+    if (isNtlm)
+    {
+       // This does not happen. But without
+       // this check, the compiler in release builds fails - isNtlm remains unused
+       return GSS_S_BAD_MECH;
+    }
+
```

Comment:
Re:

> Or, do we even need the parameter at all?

Actually it is needed for heimdal version of gssapi (OSX).  Perhaps you are suggesting that I use two functions ```NetSecurityNative_InitSecContext``` and ```NetSecurityNative_InitSecContextNtlm``` ?